mypy/subtypes.py

```diff
@@ -405,7 +406,22 @@ def visit_none_type(self, left: NoneType) -> bool:
                 members = self.right.type.protocol_members
                 # None is compatible with Hashable (and other similar protocols). This is
                 # slightly sloppy since we don't check the signature of "__hash__".
-                # None is also compatible with `SupportsStr` protocol.
+                # None is also compatible with `SupportsStr` and `SupportsBool` protocols.
+                if self.right.type.defn.info and "__bool__" in self.right.type.defn.info.names:
+                    bool_method = self.right.type.defn.info.names["__bool__"]
```

Comment:
I think that instead we should try changing `None` to have `def __bool__(self) -> Literal[False]` signature.

Reply:
You mean the `mypy.types.NoneType`?

Reply:
Yes, the member `__bool__` of `NoneType`